gprofiler/profilers/node.py

```diff
@@ -94,80 +105,79 @@ def _get_debugger_url() -> str:
     return cast(str, response_json[0]["webSocketDebuggerUrl"])
 
 
-def _send_socket_request(sock: WebSocket, cdp_request: Dict) -> None:
-    sock.send(json.dumps(cdp_request))
-    message = sock.recv()
-    try:
-        message = json.loads(message)
-    except json.JSONDecodeError:
-        raise NodeDebuggerUnexpectedResponse(message)
-
-    if (
-        "result" not in message.keys()
-        or "result" not in message["result"].keys()
-        or "type" not in message["result"]["result"].keys()
-        or message["result"]["result"]["type"] != "boolean"
-    ):
-        raise NodeDebuggerUnexpectedResponse(message)
-
-
-def _execute_js_command(sock: WebSocket, command: str) -> Any:
+@retry(NodeDebuggerProcessUndefined, 5, 0.5)
+def _evaluate_js_command(sock: WebSocket, command: str, expected_result: int) -> Any:
+    # Check if process or process.mainModule in command, and if it is, check if it is defined in js context
+    if "process.mainModule" in command:
+        command = f'typeof(process.mainModule) === "undefined" ? "process undefined" : {command}'
+    if "process" in command:
+        command = f'typeof(process) === "undefined" ? "process undefined" : {command}'
```

Comment:
So for `process.mainModule` we first check if `process` is defined and THEN check if `process.mainModule` is defined. Ay?

Reply:
Exactly